### PR TITLE
Fix multi screen timeouts

### PR DESCRIPTION
**Pull request by notimaginative**
Attempt to fix any states that cause multi to time out or that are poorly handled in multi.

### CODE REVIEW

**Comment by notimaginative** [comment]
Work in progress. More to come as we find and fix issues...

File: code/cutscene/movie.cpp [hunk 1]
```diff
@@ -10,6 +10,7 @@
 
 
 #ifdef _WIN32
+#define WIN32_LEAN_AND_MEAN
```
Comment (JohnAFernandez) [inline code/cutscene/movie.cpp:13]: What's this for? I didn't see a reference to it in the changes.

File: code/popup/popupdead.cpp [hunk 2]
```diff
@@ -445,65 +445,67 @@ int popupdead_do_frame(float  /*frametime*/)
...
+
+				case POPUPDEAD_RESPAWN_ONLY:
+				case POPUPDEAD_RESPAWN_QUIT:
+					Popupdead_choice = POPUPDEAD_DO_RESPAWN;
+					if ( (Popupdead_multi_type == POPUPDEAD_RESPAWN_QUIT) && (choice == 1) )
+						Popupdead_choice = POPUPDEAD_DO_MAIN_HALL;
+					break;
+
+				default:
+					Int3();
```
Comment (JohnAFernandez) [inline code/popup/popupdead.cpp:476]: While we're here, we should change this to an `UNREACHABLE()`

File: code/popup/popupdead.cpp [hunk 3]
```diff
@@ -445,65 +445,67 @@ int popupdead_do_frame(float  /*frametime*/)
...
```
Comment (JohnAFernandez) [inline code/popup/popupdead.cpp:506]: UNREACHABLE()

**Review by JohnAFernandez — CHANGES_REQUESTED** [review]
One question and two suggested changes.  Unreachable can make it a lot easier for users to know when to report something to SCP.

**Comment by JohnAFernandez** [comment]
Also, this does not require a multi bump?

File: code/cutscene/movie.cpp (hunk 1 above)
Comment (notimaginative) [reply code/cutscene/movie.cpp:13]: That prevents `windows.h` from including the winsock headers, which conflict with including them in psnet by way of `multi.h`.

Or as I like to say: "I hate Windows..." ;)

File: code/popup/popupdead.cpp (hunk 2 above)
Comment (notimaginative) [reply code/popup/popupdead.cpp:476]: Ah, wasn't aware of that macro. I'll get that added.

**Comment by notimaginative** [comment]
No multi bump required here. The state changes are discarded without error on older builds and none of the other changes affect compatibility.

**Review by JohnAFernandez — COMMENTED** [review]
Looks even better than before!  I saw the extra safety checks.  Instead of approving, as discussed on discord, I'll await the last timing changes.

**Review by JohnAFernandez — APPROVED** [review]
Shivansps just reported *no* issues with this build.  Time to merge.

**Pull request merged**